### ISSUE TITLE
Arabic Letters With Figo Font are Disjoined 

### ISSUE THREAD

**Issue by MohieEldinMuhammad** [pymupdf/PyMuPDF#1719]
After Following this issue **TextWriter.writeText : Arabic texts are saved in the document from left to right instead of right-left #897**
i was able to write Arabic but the letters are disjointed 

```
page.insert_textbox( bbox ,translated_text[::-1] ,fontsize= 10 ,fontname= "figo",align="r")
```

![image](https://user-images.githubusercontent.com/89604782/169288723-951cfcfd-4a82-4064-8e00-d4049530fc03.png)

 - Operating system: Working on google colab
 - Python version: Latest 
 - PyMuPDF version: !pip install PyMuPDF

Any ideas on what will make the letters joined as usual in Arabic. For Example i need  --> محتويات instead of م ح ت و ي ا ت

**Comment by JorjMcKie** [comment]
Not a bug!
You did not use TextWriter, but the older `page.insert_textbox()`.

**Comment by MohieEldinMuhammad** [comment]
@JorjMcKie I also used it like this 
```
text = "محيي الدين محمد"
font = fitz.Font("figo")
doc = fitz.open()
page = doc.new_page()
text_writer = fitz.TextWriter(page.rect)
text_writer.fill_textbox((50,100,100,200), text, font=font,right_to_left=True)
text_writer.write_text(page)
doc.ez_save("test.pdf")

```
but the same problem

**Comment by JorjMcKie** [comment]
If the single letters appear in the correct, Arabic sequence - do they!?, I fear I cannot help you.
I have no other option available to take influence.
Maybe a later PyMuPDF version can bring relief - please see [this](https://github.com/pymupdf/PyMuPDF/discussions/1029) highly interesting post.

**Comment by JorjMcKie** [comment]
I researched a little more and found confirmation:
What you want is not supported yet - in neither PyMuPDF nor MuPDF.
It will be possible only after integrating a so-called "text shaping engine" in MuPDF, which is announced for their version 1.21.0 later this year.
After extending PyMuPDF to supporting this MuPDF version, your requirements can be met.
So for now I am closing this issue with marking it "postponed".

**Comment by rajpatel2240-odoo** [comment]
@JorjMcKie any update on this?

**Comment by JorjMcKie** [comment]
> @JorjMcKie any update on this?

Sorry, no news here.

**Comment by JorjMcKie** [comment]
The coming version 1.23.8 will include a new Page method ´insert_htmlbox()´, which will be usable similar to `Page.insert_textbox()` or `TextWriter.fill_textbox()`.
The difference is that arbitrary languages, scripts, fonts etc can be contained in the text. Optionally, the text can also contain HTML / CSS language pieces.
The output will be text-shaped using the HarfBuzz library, thus covering a wide range of these situations.
